eskipfile/remote.go

```diff
@@ -135,6 +138,19 @@ func (client *remoteEskipFile) LoadUpdate() ([]*eskip.Route, []string, error) {
 	return newRoutes, deletedRoutes, err
 }
 
+func (client *remoteEskipFile) Close() {
+	if client != nil {
+		client.once.Do(func() {
+			if client.http != nil {
+				client.http.Close()
+			}
+			if client.eskipFileClient != nil {
+				client.eskipFileClient.Close()
+			}
+		})
+	}
+}
```

Comment:
Do we need nil checks here?
- nil *remoteEskipFile seems to be weird, I know go allows nil target object but I think by this check we may hide some kind of underlying problem
- client.http and client.eskipFileClient could not be nil afaict

Reply:
I think if we cleanup (Close) the remoteEskipFile it should be nil, so every concurrent access would be nil ptr.
The client ones I will clean up, I agree.

Reply:
Hm, we use `client.once.Do` which should take care of concurrent access

Reply:
but if `client` is nil how can it figure this out?

Reply:
How client can be nil in the first place?

Reply:
I mean on tear down. You have 2 goroutines one is entering `once` and the other is blocked. After cleanup `client` is nil but you have a blocked goroutine that will be woken up to check `client.once.Do` and then you deref the nil ptr. This is just a wild guess, because of the test output.

Reply:
I do not see where we set client to nil

Reply:
True we don't, but it also does not matter right now to do more checking than maybe needed. It's not the hot path and we can also make it "nice" later or is this for you a hard mandatory fix that we need to do now?

Reply:
This is a new method where we add != nil check for no reason. As a reader I am confused by this check - it hints that client can be nil while it can not. Why fix it later if we can just not add it? :shrug:

Reply:
I've removed nil check